src/App.tsx

```diff
@@ -1,19 +1,66 @@
-import { useState, useEffect, useRef } from 'react'
+import { useState, useEffect, useRef, useCallback } from 'react'
+import { DndProvider, useDrag, useDrop } from 'react-dnd'
+import { HTML5Backend } from 'react-dnd-html5-backend'
 import './App.css'
 
 interface Todo {
-  id: number;
+  id: string;
   text: string;
   completed: boolean;
   dueDate: Date;
 }
 
+interface DraggableItemProps {
+  id: string;
+  index: number;
+  moveItem: (dragIndex: number, hoverIndex: number) => void;
+  children: React.ReactNode;
+}
+
+const DraggableItem: React.FC<DraggableItemProps> = ({ id, index, moveItem, children }) => {
```

Comment:
I'm seeing some weird behaviour here when I have items in the 'today' group and the 'tomorrow' group. If I drag something from 'today', then I can't move it within the today group, but when I hover over 'tomorrow' the items there move. I think they should just be separate groups for now.